src/Provider/3dTilesProvider.js

```diff
@@ -62,6 +62,21 @@ function $3dTilesIndex(tileset, baseURL) {
 function preprocessDataLayer(layer, view, scheduler) {
     layer.sseThreshold = layer.sseThreshold || 16;
     layer.cleanupDelay = layer.cleanupDelay || 1000;
+    layer.metadataToElements = (meta) => {
```

Comment:
`metadataToElements`  shouldn't be exposed in layer. We must avoid putting processing in the provider (especially that the providers should disappear)

why not do it in the  `attachedLayer.update`?

Reply:
> metadataToElements shouldn't be exposed in layer
> We must avoid putting processing in the provider

Why?

>  (especially that the providers should disappear)

I don't think so. While currently their scope/responsability is unclear, I don't think they'll ever disappear. For instance, in the 3dtiles case you'll always need "something" that understands the tileset.json structure and parse/prepare it for the processing code to operate on.

The purpose of this `metadataToElements` function is to allow Processing code to be independant of the providers.

Reply:
We already had the same discussion for the picking feature where no good conclusion was reached. 

This confirms that the Provider/Processing boundary isn't good enough, but IMHO this shouldn't prevent merging this.

Reply:
As presented in #798, this PR proposes, not to couple the provider and the processing, as it makes the architecture more rigid and less modular. We must try to place this code in the processing not to increase this coupling

Reply:
You missed the point again and again.

I won't repeat myself please read my previous comments.